### ISSUE TITLE
Trip Planner

### ISSUE THREAD

**Issue by epicstar** [rectangle-dbmi/Realtime-Port-Authority#220]
Self explanatory

**Comment by epicstar** [comment]
Google Places API is a nice start

**Comment by epicstar** [comment]
Seems we could use the Port Authority trip planner... let's see if they will open it

**Comment by epicstar** [comment]
The implications of this is that we will need to find a workable search space to connect two points to each other via bus routes.

I believe the fastest and most feasible way to search this is through bus stop metadata. Each bus stop is assigned a list of bus routes that stop on that bus stop.

I think this is how this will work:

1. find the closest bus stops on both points
2. create a set of bus routes from all bus routes that are attached to the closest stops of each point. Each point will have a set of bus routes
3. find all intersecting bus routes on both lists of bus routes for both points. The final result should give you the final list of bus routes that will directly take you from the "from-point" to the "destination point"

Limitations:
1. This will not currently work on connecting bus routes, or routes that require 2+ buses to get from the "from-point" to the "to-point"

Required data:
1. figure out how to get all relevant bus stop data into the app considering the minimum memory constraints of the phone
2. find a way to do a search to find the closest stops between two points in minimal time... preferably not O (N).

**Comment by epicstar** [comment]
Constraints here:

1. [API to get updated static data easily](http://transitfeeds.com/p/port-authority-of-allegheny-county/274)
2. [Exploration of GTFS data directly in app -- out of the question?](http://stackoverflow.com/questions/15605394/use-of-compressed-gtfs-data-in-android-app)
3. [Associating routes for a given stop in GTFS](http://stackoverflow.com/questions/31435147/how-do-i-find-the-routes-associated-with-a-stop-using-gtfs)

**Comment by epicstar** [comment]
On the note of GTFS directly on our app.... Seems we will have to process it on a server and have the mobile apps store this data on disk.

**Comment by RitwikGupta** [comment]
This could possibly be done on a micro AWS instance?

**Comment by epicstar** [comment]
Yes, this is a possibility. The only problem with AWS is that it's only free for a year.

**Comment by epicstar** [comment]
I just pushed a gtfs-prototype that takes in a zipped gtfs file (~5MB) and creates a stop to route mapping. The process on my computer took 3.5 seconds and the output file is only 1.7 MB to store as an unminified json with the aid of the OneBusAway API. This seems to be completely reasonable to store this data in a java object on an android phone. However, I'll be interested to see if the OneBusAway library will actually work on an android device.
